tests/test_odl.py

```diff
@@ -1925,7 +1932,493 @@ def canonicalize_author_name(self, identifier, working_display_name):
+            self._db,
+            collection=collection,
+            http_get=http_get,
+        )
+
+        return importer
+
+    def _get_test_feed(self, licenses: List[TestLicense]) -> str:
+        """Get the test ODL feed with specific licensing information.
+
+        :param licenses: List of ODL licenses
 
+        :return: Test ODL feed
+        """
+        env = Environment(
+            loader=FileSystemLoader(self.ODL_TEMPLATE_DIR), autoescape=select_autoescape()
+        )
+        template = env.get_template(self.ODL_TEMPLATE_FILENAME)
+        feed = template.render(licenses=licenses)
+
+        return feed
+
+    def _import_test_feed(
+        self,
+        licenses: List[TestLicense],
+        license_infos: Optional[List[Optional[TestLicenseInfo]]] = None,
+    ) -> Tuple[List[Edition], List[LicensePool], List[Work]]:
+        """Import the test ODL feed with specific licensing information.
+
+        :param licenses: List of ODL licenses
+        :param license_infos: List of License Info Documents
+
+        :return: 3-tuple containing imported editions, license pools and works
+        """
+        feed = self._get_test_feed(licenses)
+        data_source = DataSource.lookup(self._db, "Feedbooks", autocreate=True)
+        collection = MockODLAPI.mock_collection(self._db, protocol=self.ODL_PROTOCOL)
+        collection.external_integration.set_setting(
+            Collection.DATA_SOURCE_NAME_SETTING, data_source.name
+        )
+        license_status_response = MagicMock(
+            side_effect=[
+                (200, {}, str(license_status) if license_status else "{}") for license_status in license_infos
+            ]
+            if license_infos
+            else [(200, {}, {})]
+        )
+        importer = self._create_importer(collection, license_status_response)
+
+        (
+            imported_editions,
+            imported_pools,
+            imported_works,
+            _,
+        ) = importer.import_from_feed(feed)
+
+        return imported_editions, imported_pools, imported_works
+
+
+class TestODLExpiredItemsReaperSingleLicense(TestODLExpiredItemsReaper):
+    """Class testing that the ODL 1.x reaper correctly processes publications with a single license."""
+
+    @parameterized.expand([
+        (
+            "expiration_date_in_the_past",
+            # The license expires 2021-01-01T00:01:00+01:00 that equals to 2010-01-01T00:00:00+00:00, the current time.
+            # It means the license had already expired at the time of the import.
+            TestLicense(expires=dateutil.parser.isoparse("2021-01-01T00:01:00+01:00"))
+        ),
+        (
+            "total_checkouts_is_zero",
+            TestLicense(total_checkouts=0)
+        ),
+        (
+            "remaining_checkouts_is_zero",
+            TestLicense(total_checkouts=10, concurrent_checkouts=5),
+            TestLicenseInfo(remaining_checkouts=0, available_concurrent_checkouts=0)
+        )
```

Comment:
So at this point we can just list all the ways a license might need to be reaped, and test that the reaper takes care of them.